facade/src/test/java/org/jboss/pnc/facade/providers/BuildConfigProviderTest.java

```diff
@@ -0,0 +1,258 @@
+import org.jboss.pnc.spi.datastore.repositories.api.Repository;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class BuildConfigProviderTest extends AbstractProviderTest<BuildConfiguration>{
+
+    @Mock
+    private BuildConfigurationRepository repository;
+
+    @Mock
+    private BuildConfigurationAuditedRepository buildConfigurationAuditedRepository;
+
+    @Spy
+    @InjectMocks
+    private BuildConfigurationProviderImpl provider;
+
+    @Override
+    protected AbstractProvider provider() {
+        return provider;
+    }
+
+    @Override
+    protected Repository<BuildConfiguration, Integer> repository() {
+        return repository;
+    }
+
+    private BuildConfiguration bc = prepareBuildConfig("Creative", 4, 4, 4);
+
+    @Before
+    public void fill() {
+        final BuildConfiguration a = prepareBuildConfig("First!", 1, 1, 1);
+        final BuildConfiguration b = prepareBuildConfig("Second!!", 1, 2, 2, a);
+        final BuildConfiguration c = prepareBuildConfig("THIRD!!!", 2, 2, 3, a, b);
+        final BuildConfiguration d = prepareBuildConfig("Fourth-_-", 3, 3, 3, a, b, c);
+        List<BuildConfiguration> configs = new ArrayList<>(Arrays.asList(new BuildConfiguration[] { a, b, c, d, bc }));
+        fillRepository(configs);
+    }
+
+    @Test
+    public void testGetSpecific() {
+        org.jboss.pnc.dto.BuildConfiguration buildConfiguration = provider.getSpecific(bc.getId().toString());
+
+        assertThat(buildConfiguration).isNotNull();
+        assertThat(buildConfiguration.getId()).isEqualTo(bc.getId().toString());
+        assertThat(buildConfiguration.getName()).isEqualTo(bc.getName());
+    }
+
+    @Test
+    public void testGetAll() {
+        Page<org.jboss.pnc.dto.BuildConfiguration> all = provider.getAll(0, 10, null, null);
+
+        assertThat(all.getContent())
+                .hasSize(5)
+                .haveExactly(1, new Condition<>(e -> e.getName().equals(bc.getName()), "BC Present"));
+    }
+
+    @Test
+    public void testStore() {
+        final String name = "NewHire";
+        final String repoId = "12";
+        final String envId = "12";
+        final String projId = "12";
+
+        org.jboss.pnc.dto.BuildConfiguration buildConfiguration = org.jboss.pnc.dto.BuildConfiguration.builder()
+                .name(name)
+                .buildType(BuildType.MVN)
+                .project(org.jboss.pnc.dto.Project.builder().id(projId).build())
```

Comment:
Wouldn't ProjectRef be technically better?

Reply:
Huh, ya.. you are right. Full project dto can be misleading.